packages/ubuntu_desktop_installer/lib/pages/who_are_you_page.dart

```diff
@@ -0,0 +1,328 @@
+                    ),
+                    Row(
+                      children: [
+                        Radio<LoginStrategy>(
+                            value: LoginStrategy.AUTO_LOGIN,
+                            groupValue: _loginStrategy,
+                            onChanged: (_) => setState(() {
+                                  _loginStrategy = LoginStrategy.AUTO_LOGIN;
+                                })),
+                        Padding(
+                          padding: const EdgeInsets.all(8.0),
+                          child: Text(lang.whoAreYouPageLoginStrategyAutoLogin),
+                        )
+                      ],
+                    ),
+                    Row(
+                      children: [
+                        Radio<LoginStrategy>(
+                            value: LoginStrategy.REQUIRE_PASSWORD,
+                            groupValue: _loginStrategy,
+                            onChanged: (_) => setState(() {
+                                  _loginStrategy =
+                                      LoginStrategy.REQUIRE_PASSWORD;
+                                })),
+                        Padding(
+                          padding: const EdgeInsets.all(8.0),
+                          child: Text(
+                              lang.whoAreYouPageLoginStrategyRequirePassword),
+                        )
+                      ],
+                    ),
+                  ],
+                ),
+              ),
+              actions: <WizardAction>[
+                WizardAction(
+                  label: lang.backButtonText,
+                  onActivated: Navigator.of(context).pop,
+                ),
+                WizardAction(
+                  label: lang.startInstallingButtonText,
+                  highlighted: true,
+                  onActivated: () async {
+                    if (_realNameFormKey.currentState!.validate() &&
+                        _computerNameFormKey.currentState!.validate() &&
+                        _usernameFormKey.currentState!.validate() &&
+                        _passwordFormKey.currentState!.validate() &&
+                        _confirmPasswordFormKey.currentState!.validate()) {
+                      final client =
+                          Provider.of<SubiquityClient>(context, listen: false);
+
+                      final key = encrypt.Key.fromLength(32);
+                      final iv = encrypt.IV.fromLength(16);
+                      final encrypter = encrypt.Encrypter(encrypt.AES(key));
+
+                      final encrypted =
+                          encrypter.encrypt(_passwordController.text, iv: iv);
+
+                      await client.setIdentity(IdentityData(
+                          hostname: _computerNameController.text,
+                          realname: _realNameController.text,
+                          username: _usernameController.text,
+                          cryptedPassword: encrypted.toString()));
```

Comment:
this kind of business logic would be nice to do in a view model, and test it with a mock client

Reply:
Hm, isn't the user data used also in other pages? This would be an argument for a model dir, imo